frontend/src/views/upload-project/UploadProjectView.tsx

```diff
@@ -50,37 +68,58 @@ export const UploadProjectView = () => {
         file: null,
       },
     };
+    let response: any;
     try {
-      const response = await graphQLHttpClient.sendFile(uploadProjectMutation, variables, file);
-      const action = { type: HANDLE_SUBMIT__ACTION, response };
-      dispatch(action);
+      dispatch({ type: 'HANDLE_UPLOAD' });
+      response = await graphQLHttpClient.sendFile(uploadProjectMutation, variables, file);
+      if (response?.data?.uploadProject?.message) {
```

Comment:
The GraphQL API is typed, use this type to find more about the result instead of relying on such brittle checks